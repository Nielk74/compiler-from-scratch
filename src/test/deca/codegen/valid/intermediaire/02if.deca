{
    int i = 1;
    int j = 1;
    int k = 2;

    // == 
    if (i == j) {
        println("i = j");
    }
    else {
        println("error 1");
    }

    // !=
    if (i != j) {
        println("error 2");
    }
    else {
        println("i = j");
    }

    // <
    if (i < j) {
        println("error3");
    }
    // <
    if (i < k) {
        println("i < k");
    }
    // <=
    if (i <= j) {
        println("i <= j");
    }
    // <=
    if (i <= k) {
        println("i <= k");
    }
    // >
    if (i > k) {
        println("error4");
    }
    // >=
    if (i >= k) {
        println("error5");
    }
    // !=
    if (i != k) {
        println("i != k");
    }
}