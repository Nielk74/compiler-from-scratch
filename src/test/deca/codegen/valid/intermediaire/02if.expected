i = j
i = j
i < k
i <= j
i <= k
i != k
